scalafix-core/src/main/scala/scalafix/rewrite/ExplicitReturnTypes.scala

```diff
@@ -58,14 +85,39 @@ case class ExplicitReturnTypes(mirror: Mirror)
         typ <- defnType(defn)
       } yield ctx.addRight(replace, s": ${typ.treeSyntax}")
     }.to[Seq]
+
+    import ctx.config.explicitReturnTypes._
+
+    def matchesMemberVisibility(mods: Seq[Mod]): Boolean =
+      memberVisibility.contains(visibility(mods))
+
+    def matchesMemberKind(defn: Defn): Boolean =
+      kind(defn).exists(memberKind.contains)
+
+    def matchesSimpleDefinition(rhs: Term): Boolean =
+      rhs.is[Lit] && skipSimpleDefinitions
```

Comment:
I'd say in this case we want to treat identifiers as "complex definitions" since inferring their type may require inferring complex definitions

```scala
def foo = Generic[Bar] // infers to complex type `String :+: Int :+ ... :+: HNil`
implicit val x  = foo // not simple, we want explicit type here.
```

I think literals are a good enough for "simple definitions", at least to start with. We may later want to handle more sophisticated cases like

```
val foo = Option.empty[Int]
// after rewrite
val foo: Option[Int] = None
```